### PR TITLE
Make GroupKind the key of ApplierOptions

### PR DESCRIPTION
**Pull request by timuthy**
**What this PR does / why we need it**:
This PR changes the key of the `ApplierOptions` from `Kind` to `GroupKind` which is more accurate.

/cc @adracus 

**Special notes for your reviewer**:
I also removed the [initial invalidation](https://github.com/gardener/gardener/blob/da15a37de8912afcc3c2113456f12c2b6d502f0b/pkg/client/kubernetes/apply.go#L74) of the `DiscoveryClient`. Due to https://github.com/kubernetes/kubernetes/commit/c94bee0b8b88851e5f5fd6538b99adff8b3a13f0#diff-498e117e58cba7576e99cf7fd3cb023eR129, this is not required anymore.

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/client/kubernetes/apply.go [hunk 1]
```diff
@@ -76,10 +80,6 @@ func (c *Applier) applyObject(ctx context.Context, desired *unstructured.Unstruc
 		return err
 	}
 
-	if !c.discovery.Fresh() {
```
Comment (rfranzke) [inline pkg/client/kubernetes/apply.go]: Why is this removed?

**Review by rfranzke — COMMENTED** [review]
Looks good, just a short qstn :)

**Review by rfranzke — APPROVED** [review]
/lgtm

**Review by adracus — APPROVED** [review]
